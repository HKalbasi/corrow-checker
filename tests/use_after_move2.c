#include <stddef.h>

void *owned malloc(unsigned long size);
void free(void *owned ptr);

int main()
{
    int *p1 = malloc(10);
    free(p1);
    free(p1); // use after move
    return 0;
}