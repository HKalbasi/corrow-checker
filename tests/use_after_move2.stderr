Error: Potential use after move
    ╭─[preproccessed_file.c:1:1]
    │
 32 │     free(p1);
    │          ─┬  
    │           ╰── The value is moved here
 33 │     free(p1);
    │          ─┬  
    │           ╰── The value is used here after move
────╯